be/src/exec/vectorized/partition/partition_hash_map.h

```diff
@@ -113,11 +136,19 @@ struct PartitionHashMapBase {
             }
             value.select_indexes.push_back(i);
             value.remain_size--;
+            total_num_rows++;
         }
 
         for (const auto& key : visited_keys) {
             flush(*(hash_map[key]), chunk);
         }
+
+        // The first i rows has been pushed into hash_map
+        if (is_downgrade && i > 0) {
+            for (auto& column : chunk->columns()) {
+                column->remove_first_n_values(i);
```

Comment:
shouldn't remove too often。

Reply:
It happens only once, i.e. when downgrade happens.